packages/react/src/components/header/components/headerActionBar/HeaderActionBarNavigationMenu.module.scss

```diff
@@ -1,42 +1,225 @@
-  transition: 200ms max-height;
-}
-
-.mobileMenuOpen {
-  max-height: 100vh;
+  position: absolute;
+  transition: 0.3s max-height;
+  width: 100%;
 }
 
-.headerNavigationMenuList {
+.menu {
+  display: flex;
+  flex-flow: column nowrap;
+  height: min-content;
   list-style: none;
   margin: 0;
   margin-top: -1px;
+  overscroll-behavior: contain;
   padding: 0;
+  width: 100vw;
+
+  .activeListItem {
+    border-top: none;
+    min-height: 72px;
+  }
+
+  .activeMobileLink.activeMobileLink {
+    font-size: var(--fontsize-heading-m);
+    margin: var(--spacing-m) var(--spacing-s);
+    text-decoration: underline;
+
+    &:hover {
+      text-decoration: none;
+    }
+
+    &:focus-visible {
+      border: none;
+      outline: var(--header-focus-outline-width) solid var(--header-focus-outline-color);
+      outline-offset: calc(var(--header-focus-outline-width) - (var(--header-focus-outline-width) * 2));
+    }
+  }
+
+  .hideDropdownButton {
+    display: none;
+  }
+
+  .previousListItem {
+    align-items: center;
+    display: flex;
+    min-height: auto;
+    padding: var(--spacing-m) var(--nav-mobile-margin-left) 0 var(--nav-mobile-margin-left);
+
+    .previousButton {
+      align-items: center;
+      background-color: var(--nav-button-background-color);
+      border: none;
+      color: var(--nav-link-font-color);
+      display: flex;
+      height: 32px;
+      justify-content: center;
+      width: 32px;
+    }
+  }
+
+  .previousMobileLink.previousMobileLink {
+    color: var(--nav-link-font-color);
+    font-size: var(--fontsize-body-m);
+    font-weight: 500;
+    margin-left: var(--spacing-xs);
+  }
+}
+
+.mobileDropdownLink {
+  position: relative;
+}
+
+.mobileLinkDropdown {
+  display: block;
+  left: 0;
+  position: relative;
+  right: 0;
+  top: auto;
+  width: 100%;
+}
+
+.mobileLinkDropdownButton.mobileLinkDropdownButton {
+  background-color: transparent;
+  margin: 0;
+  padding: 20px;
+  transform: rotate(-90deg);
+
+  &:hover {
+    background-color: transparent;
+  }
+
+  svg:hover {
+    transform: translateY(4px);
+    transition: transform 0.2s linear;
+  }
+
+  &:focus-visible {
+    border: none;
+    outline: var(--header-focus-outline-width) solid var(--header-focus-outline-color);
+    outline-offset: calc(var(--header-focus-outline-width) - (var(--header-focus-outline-width) * 2));
+
+    svg {
+      transform: translateY(4px);
+      transition: transform 0.2s linear;
+    }
+  }
+}
+
+.mobileLinkWrapper {
+  box-sizing: border-box;
+
+  .mobileLink.mobileLink {
+    border-right: 1px solid var(--color-black-20);
+    box-sizing: border-box;
+    flex-grow: 1;
+    margin: 0;
+    margin-left: 0;
+    padding: 17px 16px;
+    position: relative;
+
+    &:focus-visible {
+      border: none;
+      outline: var(--header-focus-outline-width) solid var(--header-focus-outline-color);
+      outline-offset: -3px;
+    }
+  }
+}
+
+.mobileMenuOpen {
+  height: 100vh;
+  max-height: 100vh;
+}
```

Comment:
If we want be fancy-pancy, then `100dvh` would be best option here. 😄 
https://ishadeed.com/article/new-viewport-units/

When checking hel.fi animations, I saw they used these new units. Support is quite good.

`100vh` is fine here too. Probably all components should support `dvh` and `dvw` and better to fix all together.

Reply:
In addition to header components, viewport units are used in CookieConsent, DatePicker and once in site layout. Not that many places, but probably could change these separately 🤔

Reply:
I agree.